lib/src/configuration.dart

```diff
@@ -729,16 +729,16 @@ class SyncSessionError extends SyncError {
   }
 }
 
-/// Network resolution error
-class SyncResolveError extends SyncError {
-  /// The numeric value indicating the type of the network resolution sync error.
-  SyncResolveErrorCode get code => SyncResolveErrorCode.fromInt(codeValue);
+/// Web socket error
+class SyncWebSocketError extends SyncError {
```

Comment:
This is a breaking change and requires 2.0 release. Should we make it backward compatible?

Reply:
We could keep the `SyncResolveError` or have `SyncWebSocketError` implement `SyncResolveError`(which it trivially do).. but I'm not sure I'm a fan of either.

On the other hand bumping to 2.0 for this minor tweak also seems excessive.

Reply:
* Deprecated `SyncResolveError` and `SyncResolveErrorCode`.
* Added `SyncWebSocketError` and `SyncWebSocketErrorCode` for web socket connection sync errors.